src/node/interfaces.cpp

```diff
@@ -828,20 +828,20 @@ class ChainImpl : public Chain
         // Now dump value to disk if requested
         return *action == interfaces::SettingsAction::SKIP_WRITE || args().WriteSettingsFile();
     }
-    bool overwriteRwSetting(const std::string& name, common::SettingsValue value, bool write) override
+    bool overwriteRwSetting(const std::string& name, common::SettingsValue value, interfaces::SettingsAction action) override
     {
-        if (value.isNull()) return deleteRwSettings(name, write);
+        if (value.isNull()) return deleteRwSettings(name, action);
         return updateRwSetting(name, [&](common::SettingsValue& settings) {
             settings = std::move(value);
-            return write ? interfaces::SettingsAction::WRITE : interfaces::SettingsAction::SKIP_WRITE;
+            return action;
         });
     }
-    bool deleteRwSettings(const std::string& name, bool write) override
+    bool deleteRwSettings(const std::string& name, interfaces::SettingsAction action) override
     {
         args().LockSettings([&](common::Settings& settings) {
             settings.rw_settings.erase(name);
         });
-        return !write || args().WriteSettingsFile();
+        return action == interfaces::SettingsAction::SKIP_WRITE || args().WriteSettingsFile();
```

Comment:
I guess you are trying to avoid a situation where we would need to write:

```
return action == interfaces::SettingsAction::SKIP_WRITE || action == interfaces::SettingsAction::NEW_ACTION || args().WriteSettingsFile();
```

In some cases we would still need to modify this to handle the new action if it has additional effects beyond just writing, such as writing and logging, for example.

---
I've updated to your suggestion, as it is more maintainable.